titus-server-master/src/test/java/com/netflix/titus/master/clusteroperations/ClusterAgentAutoScalerTest.java

```diff
@@ -185,7 +188,7 @@ public void testScaleUpForFailingTasks() {
         Map<TaskPlacementFailure.FailureKind, Map<String, List<TaskPlacementFailure>>> taskPlacementFailures = createTaskPlacementFailures(ImmutableMap.of(
                 AllAgentsFull, 10
         ), Tier.Flex);
-        when(schedulingService.getLastTaskPlacementFailures()).thenReturn(taskPlacementFailures);
+        doReturn(taskPlacementFailures).when(schedulingService).getLastTaskPlacementFailures();
```

Comment:
Was the `thenReturn()` type checking getting confused?

Reply:
Yes, this is the way you are supposed to use Mockito with generic types/methods.